### PR TITLE
Retrieve and update an incomplete application via the organisation page

### PR DESCRIPTION
**Pull request by simonnineteen90**
https://eaflood.atlassian.net/browse/SFI-76

### CODE REVIEW

File: app/agreement.js [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 
 const getAgreementBySbi = async (sbi) => {
   return db.agreement.findAll({
-    raw: true,
+    include: [{ model: db.status, as: 'status'}],
     where: { sbi }
   })
 }
 
 const getAgreement = async (agreementNumber, sbi) => {
```
Comment (marctemp) [inline app/agreement.js:16]: Could this be named to reflect it's by SBI and Agreement Number (<-- if it is)

File: app/agreement.js [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+  })
 }
 
 const getAgreementBySbi = async (sbi) => {
   return db.agreement.findAll({
-    raw: true,
+    include: [{ model: db.status, as: 'status'}],
     where: { sbi }
   })
 }
```
Comment (marctemp) [inline app/agreement.js:14]: Also, dis shizz is cooowwll af 8)

**Comment by marctemp** [comment]
Great code guys @pxp420 and @simonnineteen90, looks great!

Only have some minor suggestions left; need to run the repo assa whole but I'm heading off now so will pick this back up on Monday -- enjoy your weekends!

File: app/agreement.js [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-    raw: true,
+    include: [{ model: db.status, as: 'status'}],
     where: { agreementNumber, sbi }
   })
 }
 
 const addAgreement = async (agreement) => {
   const { agreementNumber, crn, organisation } = agreement
-  await db.agreement.create({ agreementNumber, crn, sbi: organisation.sbi, agreementData: agreement })
+  await db.agreement.create({ agreementNumber, crn, statusId: agreement.statusId ?? 1, sbi: organisation.sbi, agreementData: agreement })
```
Comment (marctemp) [inline app/agreement.js:25]: Could create the status model as done in `apply-web` to get rid of magic number

**Review by marctemp — CHANGES_REQUESTED** [review]
Pipeline fails on audit stage, I had this on my last PR:

- run `npm audit fix` in the repo (this will update package versions in `package-lock.json`)
- it's likely (mine did anyway) it will fail Synk test, remove `package-lock.json` and re-run `npm update` to recreate it

**Review by marctemp — APPROVED** [review]

File: app/agreement.js (hunk 3 above)
Comment (pxp420) [reply app/agreement.js:25]: Will leave how it currently is, the other method was using magic numbers as well. A good note for future work on this though.

**Pull request merged**